### PR TITLE
Add explicit compiler and stdlib requirement

### PR DESCRIPTION
**Pull request by peterjc**
See https://conda-forge.org/news/2024/01/29/rust-175-osx-sdk/

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by peterjc** [comment]
@conda-forge-admin, please rerender

**Comment by peterjc** [comment]
Let's try adding an explicit compiler requirement - see #3

**Comment by peterjc** [comment]
TravisCI worked for ARM but failed for ppc64le, specifically ``CONFIG=linux_ppc64le_ UPLOAD_PACKAGES=True PLATFORM=linux-ppc64le DOCKER_IMAGE=quay.io/condaforge/linux-anvil-ppc64le``, no sign of a log - maybe I left it too long before checking the details?

**Comment by wolfv** [comment]
I tried to trigger a rerun.

**Comment by peterjc** [comment]
Sadly I don't think that rerun request worked, still pointing at https://app.travis-ci.com/github/conda-forge/xsv-feedstock/jobs/623091067 which is over a week ago now.

But you're right that this could have been a transient failure worth retrying.

**Comment by rvalieris** [comment]
I just re-triggered the build now, looks like it passed.

Is this PR replacing the ARM migrator PR ?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,9 +9,12 @@ source:
...
 build:
-  number: 0
+  number: 1
 
 requirements:
+  build:
+    - {{ compiler("rust") }}
+    - {{ stdlib("c") }}
   host:
     - rust
```
Comment (wolfv) [inline recipe/meta.yaml]: you should be able to remove `rust` here when you have it in the build requirements.

**Review by wolfv — CHANGES_REQUESTED** [review]

**Comment by peterjc** [comment]
I’m hoping this clerical update will help the macOS ARM update to pass…

**Comment by rvalieris** [comment]
is this ok to merge then ?

**Comment by peterjc** [comment]
Yes please

**Pull request merged**